ui/src/errors.ts

```diff
@@ -24,3 +25,16 @@ export class NotImplementedError extends Error {
         super(localize('notImplementedError', '"{0}" is not implemented on "{1}".', methodName, obj.constructor.name));
     }
 }
+
+export class NoResouceFoundError extends Error {
+    constructor(errorMessage?: string, context?: IActionContext) {
```

Comment:
Just pass in `context: ITreeItemPickerContext` as the only arg and make it required. Should simplify the code

Reply:
I thought about it. But this will support more scenarios like, a caller to throw this error without an action context. The message and context can be independent.

Reply:
So `index.d.ts` is essentially the public contract of this package. If a class is not listed in there, it can't be used outside of this package. Since `NoResouceFoundError` is only used _within_ the package, I prefer to go for the simpler route. If you wanted to consume this class outside of the package (which requires listing it in `index.d.ts`), then I agree it could be worth the added complexity

Reply:
Not a blocker, though. Approved

Reply:
It should stay within the package. Wouldn't separating the parameter be more open and allows more flexibility?

Reply:
Right I'm saying I vote for simplicity if it's internal and flexibility if it's used externally